client/components/pages/files/FilesTable.vue

```diff
@@ -26,43 +45,107 @@ provide(FileTableKey, {
 provide(FileTableContextMenuKey, contextMenu)
+
+function resetPage() {
+  page.value = 0
+  pageSize.value = 10
+  pageDefinition.value = {}
+  refresh()
+}
+
+function onPage(event: DataTablePageEvent) {
+  page.value = event.page
+  pageSize.value = event.rows
+}
+
+function onSort(event: DataTableSortEvent) {
+  pageDefinition.value = {
+    ...pageDefinition.value,
+    sorter: event.multiSortMeta?.map((sort: DataTableSortMeta) => ({
+      field: sort.field,
+      direction: sort.order === 1 ? 'ASC' : 'DESC',
+    })),
+  }
+}
+
 function onRowContextMenu(event: DataTableRowContextMenuEvent) {
   contextMenu.value?.show(event.originalEvent)
 }
+
 function onRowDoubleClick(event: DataTableRowDoubleClickEvent) {
   window.open(`/${event.data.code}`)
 }
 </script>
 
 <template>
   <FilesTableContextMenu />
+
+  <div v-if="error" class="h-full w-full flex flex-col justify-center items-center">
+    <!-- TODO: Error state illustration -->
+    <img class="w-10em" src="/assets/img/new_file.png" alt="Error file">
+    <h1 class="text-2xl">
+      {{ t('pages.files.error.title') }}
+    </h1>
+    <p class="pb-10">
+      {{ t('pages.files.error.description') }}
+    </p>
+    <PButton
+      v-if="errorStatus === 400" :loading="pending" rounded :label="t('pages.files.error.reset_button')"
+      @click="resetPage()"
+    />
+    <PButton v-else :loading="pending" rounded :label="t('pages.files.error.retry_button')" @click="refresh()" />
+  </div>
+
+  <div v-else-if="files.length === 0 && !pending" class="h-full w-full flex flex-col justify-center items-center">
+    <!-- TODO: Empty state illustration -->
+    <img class="w-10em" src="/assets/img/new_file.png" alt="New file">
+    <h1 class="text-2xl">
+      {{ t('pages.files.empty.title') }}
+    </h1>
+    <p class="pb-10">
+      {{ t('pages.files.empty.description') }}
+    </p>
+    <PButton rounded :label="t('pages.files.empty.upload_button')" />
+  </div>
+
   <PDataTable
+    v-else
     v-model:selection="selectedRows"
     v-model:contextMenuSelection="selectedRow"
-    :value="files"
+    class="h-full"
     data-key="id"
+    lazy
+    :value="loading ? Array.from({ length: rows }) : files"
     scrollable
-    scroll-height="100%"
+    scroll-height="flex"
+    paginator
+    paginator-template="FirstPageLink PrevPageLink CurrentPageReport NextPageLink LastPageLink RowsPerPageDropdown"
+    :current-page-report-template="t('pages.files.table.paginator', {}, { escapeParameter: true })"
+    :rows-per-page-options="[10, 25, 50, 100]"
+    :rows="rows"
+    :total-records="totalRecords"
     sort-mode="multiple"
     removable-sort
     context-menu
-    class="h-full"
+    @page="onPage"
+    @sort="onSort"
     @row-contextmenu="onRowContextMenu"
     @row-dblclick="onRowDoubleClick"
   >
     <PColumn style="width: 3.375em;" selection-mode="multiple" />
 
     <PColumn style="width: 6em;" field="code" :header="t('pages.files.table.preview')" :sortable="false">
```

Comment:
you're not forced to do it, but maybe all those columns can be factorized generated with a clean v-for

Reply:
Meh, since everything is very different, I don't feel like doing it for now..

Reply:
fine, i'll do it myself later probably